src/ess/reduce/widgets/_base.py

```diff
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+import warnings
+from typing import Any, Protocol, runtime_checkable
+
+from ipywidgets import Widget
+
+
+@runtime_checkable
+class WidgetWithFieldsProtocol(Protocol):
+    def set_fields(self, new_values: dict[str, Any]) -> None: ...
+
+    def get_fields(self) -> dict[str, Any]: ...
```

Comment:
I assumed that `set_fields` and `get_fields` should always exist in pairs
but let me know if there's any use-case that can only have one of them but not the other.